### PR TITLE
Update theme documentation

### PR DESCRIPTION
**Pull request by rurikoaraki**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Reorganize and add some more detailed examples for theming. Updated examples to align with latest framework.

Aim is to make the documentation easier to use for people who just want FURN theming and to hook them into stock RN components.

# Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: docs/pages/Theming/CustomTheme.md [hunk 1]
```diff
@@ -2,139 +2,174 @@
...
+## Creating a custom theme
 
-## Different aspects of customization
+There are two ways to make a custom theme: you can tack onto an existing one using `ThemeRecipes`, or create one from scratch. Generally it's easier to extend one of the default ones we provide.
 
-There's two main ways to customize a FURN theme - you can specify how to customize components, or you can customize theme tokens directly by overriding certain tokens.
+### Extending an existing `ThemeReference`
 
-### Components
+We have a concept of `ThemeRecipes` which allow for layering of partial theme objects to create the ultimately desired theme. `ThemeRecipes` are functions which take a `Theme` and spit out a `PartialTheme`, which is then deep merged into the base them object.
```
Comment (samuelfreiberg) [inline docs/pages/Theming/CustomTheme.md:11]: Is there a definition of a PartialTheme anywhere that we can link? I think it's pretty self-explanatory, but I'm not 100% sure I know everything about it

Comment (rurikoaraki) [reply docs/pages/Theming/CustomTheme.md:11]: Yeah I can link to the theme.types page here, thanks!

File: docs/pages/Theming/CustomTheme.md [hunk 2]
```diff
@@ -2,139 +2,174 @@
...
-```
+### `ThemeReference` from scratch
 
-You can see what settings can be customized by looking at a component's settings, styling, or tokens file.
+You can create your own `ThemeReference` and pass it into the `ThemeProvider`. To create a `ThemeReference` you'll need to create a new instance of `ThemeReference`. You can find the type definition [here](https://github.com/microsoft/fluentui-react-native/blob/master/packages/framework/theme/src/themeReference.ts).
 
-The advantage of this approach is that the customizations are targetted to a particular component, so the changes are localized.
+## Theme property customization
+
+There's two ways to customize a FURN theme's properties. The main way is to customize theme tokens directly by overriding certain tokens. You can also specify how to customize FURN components for the whole theme.
```
Comment (samuelfreiberg) [inline docs/pages/Theming/CustomTheme.md:44]: Super Nit Opinion: When you have a certain # of choices, I always prefer a numbered list. It helps break up the text into sections and catches the eye easier, and makes a bigger impact to the reader. 

This is a personal preference, but I thought I'd mention it. Feel free to disregard if you don't agree. For example,

_There are two ways to customize a FURN theme's properties:
1. Customize theme tokens directly by overriding certain tokens (main way).
2. Specify how to customize FURN components for the whole theme._

Comment (rurikoaraki) [reply docs/pages/Theming/CustomTheme.md]: Thanks, that makes sense. Implemented this.

**Review by samuelfreiberg — APPROVED** [review]

**Pull request merged**